flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -688,4 +669,55 @@ public static Configuration adjustConfigurationForLocalExecution(final Configura
 
 		return modifiedConfig;
 	}
+
+	public static Configuration adjustConfigurationForStandaloneSetups(final Configuration configuration) {
+		return getConfigurationMapLegacyTaskManagerHeapSizeToConfigOption(
+			configuration, TaskManagerOptions.TOTAL_FLINK_MEMORY);
+	}
+
+	public static Configuration adjustConfigurationForContainerizedSetups(final Configuration configuration) {
+		return getConfigurationMapLegacyTaskManagerHeapSizeToConfigOption(
+			configuration, TaskManagerOptions.TOTAL_PROCESS_MEMORY);
+	}
+
+	private static Configuration getConfigurationMapLegacyTaskManagerHeapSizeToConfigOption(
+		final Configuration configuration, ConfigOption<MemorySize> configOption) {
+
+		final Optional<MemorySize> legacyHeapSizeOptional = getLegacyTaskManagerHeapMemoryIfExplicitlyConfigured(configuration);
+		if (legacyHeapSizeOptional.isPresent()) {
+			final Configuration copiedConfig = new Configuration(configuration);
+			copiedConfig.set(configOption, legacyHeapSizeOptional.get());
+			return copiedConfig;
+		}
+
+		return configuration;
```

Comment:
This could be expressed slightly more succinct:

```
getLegacyTaskManagerHeapMemoryIfExplicitlyConfigured(null).map(legacyHeapSize -> {
			final Configuration copiedConfiguration = new Configuration(configuration);
			copiedConfiguration.set(configOption, legacyHeapSize);
			return copiedConfiguration;
		}).orElse(configuration);
```